src/server/scripts/Northrend/CrusadersColiseum/TrialOfTheChampion/boss_argent_challenge.cpp

```diff
@@ -104,321 +150,710 @@ enum Spells
+                            lightWielder->GetMotionMaster()->MovePoint(0, lightWielder->GetHomePosition());
+                        }
                     }
-                    uiRenewTimer = urand(15000, 17000);
-                } else uiRenewTimer -= uiDiff;
+
+                    if (Creature* priestess = me->FindNearestCreature(NPC_PRIESTESS, 15.0f))
+                    {
+                        if (priestess->GetFollowAngle() != 0.f)
+                        {
+                            priestess->GetMotionMaster()->Clear();
+                            priestess->GetMotionMaster()->MovePoint(0, priestess->GetHomePosition());
+                        }
+                    }
+                }
             }
+        }
 
-            if (!bHealth && !HealthAbovePct(25))
+        void JustSummoned(Creature* summon) override
+        {
+            if (me->GetEntry() == NPC_PRIESTESS)
+                _fountainGuid = summon->GetGUID();
+        }
+
+        void DamageTaken(Unit* /*attacker*/, uint32& damage) override
+        {
+            if (me->GetEntry() == NPC_ARGENT_MONK && damage >= me->GetHealth() && !_shielded)
             {
-                me->InterruptNonMeleeSpells(true);
-                DoCastAOE(SPELL_HOLY_NOVA, false);
-                DoCast(me, SPELL_SHIELD);
-                DoCastAOE(SPELL_SUMMON_MEMORY, false);
-                DoCastAOE(SPELL_CONFESS, false);
+                _shielded = true;
+                damage = me->GetHealth() - 1;
+                me->AttackStop();
+
+                DoCast(SPELL_DIVINE_SHIELD);
+                DoCast(SPELL_FINAL_MEDITATION);
+            }
+        }
+
+        void JustDied(Unit* /*killer*/) override
+        {
+            _instance->SetData(DATA_ARGENT_SOLDIER_DEFEATED, _instance->GetData(DATA_ARGENT_SOLDIER_DEFEATED) + 1);
+            if (me->GetEntry() == NPC_PRIESTESS)
+            {
+                if (Creature* fountain = ObjectAccessor::GetCreature(*me, _fountainGuid))
+                    fountain->DisappearAndDie();
+            }
+        }
 
-                bHealth = true;
+        void SetData(uint32 type, uint32 /*data*/) override
+        {
+            _waypointId = type;
+            Position homePos;
+            GenerateFinalPosition(homePos);
+
+            if (homePos.IsPositionValid())
+            {
+                me->SetHomePosition(homePos);
+                me->SetWalk(true);
+                if (me->GetEntry() == NPC_ARGENT_MONK)
+                    me->GetMotionMaster()->MovePoint(POINT_PREFIGHT, homePos);
+            }
+        }
+
+        void JustEngagedWith(Unit* /*who*/) override
+        {
+            switch (me->GetEntry())
+            {
+            case NPC_ARGENT_LIGHWIELDER:
+                _events.ScheduleEvent(EVENT_BLAZING_LIGHT, 10 * IN_MILLISECONDS);
+                _events.ScheduleEvent(EVENT_CLEAVE, urand(4 * IN_MILLISECONDS, 6 * IN_MILLISECONDS));
```

Comment:
`EVENT_CLEAVE, 4s, 6s`

_(that applies to all eventmap calls, I'll stop mentioning them all individually now)_